docs/ts/primitives/streaming-apis.mdx

```diff
@@ -8,118 +8,269 @@ lang: ts
+When you connect to a streaming API endpoint, the client and server will do a handshake in the form of a HTTP request. If the server accepts the handshake request, a stream is returned to the client and to the API handler. Under the hood the stream is a WebSocket that can be used to send and receive messages over.
 
-Path parameters, query parameters and headers can be passed via the handshake request. The stream returned to the client and to the API handler are typed with the incoming and outgoing message types that you specify in your api.
+Path parameters, query parameters and headers can be passed via the handshake request. The stream returned to the client and to the API handler are typed with the incoming and outgoing message types that you specify in your API.
 
 ## Defining streaming APIs
 
-Similar to how you can define [RESTful API endpoints](/docs/ts/primitives/apis) with Encore, you can also easily define type-safe streaming API endpoints. They accept a handshake type, an incoming and an outgoing message type (depending on your choice of stream direction).
+Similar to how you can define [RESTful API endpoints](/docs/ts/primitives/apis) with Encore, you can also easily define type-safe streaming API endpoints. They accept a handshake type, an incoming and an outgoing message type (depending on your choice of stream direction). The type parameters are required for Encore to understand your API.
 
 If you don't need any data from the handshake, you can ignore that type, and only specify the incoming and outgoing message types.
 
-For example, if you want to have a stream of messages from server to client for a specific chat room, you could setup an `api.streamOut`:
+<GitHubLink
+    href="https://github.com/encoredev/examples/tree/main/ts/streaming"
+    desc="Example showcases the all different streaming APIs: api.streamIn, api.streamOut, and api.streamInOut"
+/>
+
+### StreamIn
+
+Use `api.streamIn` when you want to have a stream from client to server, for example if you are uploading something from the client to the server:
 
 ```typescript
+import { api } from "encore.dev/api";
+
+// Used to pass initial data, optional.
 interface Handshake {
-  roomId: string;
+  user: string;
 }
 
+// What the clients sends over the stream.
 interface Message {
-  text: string;
-  sender: string;
+  data: string;
+  done: boolean;
 }
 
-// messageStream streams out all chat messages for a certain id
-export const messageStream = api.streamOut<Handshake, Message>(
-  { path: "/chat/:roomId", expose: true },
-  async (handshake, stream) => {
-    // Use `handshake.roomId` to fetch messages
-    for await (const msg of messages) {
-      await stream.send(msg);
+// Returned when the stream is done, optional.
+interface Response {
+  success: boolean;
+}
+
+export const uploadStream = api.streamIn<
+  Handshake,
+  Message,
+  Response
+>({ path: "/upload", expose: true }, async (handshake, stream) => {
```

Comment:
```suggestion
>({ path: "/upload", expose: true }, async (handshake, stream): Promise<Response> => {
```
I think you can add the return type here

Reply:
It's not needed since the generic parameters are passed in (`api.streamIn<A, B, C>`), so I think it's cleaner this way